subsys/spm/spm.c

```diff
@@ -567,3 +582,93 @@ void spm_config(void)
 	spm_config_sram();
 	spm_config_peripherals();
 }
+
+#if CONFIG_SPM_NS_DEBUG_LOGS_DUMP
+
+static spm_ns_on_fatal_error_t ns_on_fatal_handler;
+
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_set_ns_fatal_error_handler(spm_ns_on_fatal_error_t handler)
```

Comment:
I'm thinking maybe this should go into secure_services.c.
I think this can fail if SECURE_SERVICES is turned off, since CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS (in NS app) and CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS (in SPM) need to be set for nonsecure entry functions to work.
If you make this a secure service, you don't have to worry about that.

Reply:
moved to secure services.